src/components/Header.jsx

```diff
@@ -1,69 +1,38 @@
-import { useState, useEffect } from "react";
 import { Menu } from "antd";
 import { useNavigate } from "react-router-dom";
 import "./header.css";
+import { useAuth } from "../../utils/useAuth";
+import PropTypes from "prop-types";
+import { useEffect } from "react";
 
 const Header = () => {
-  const [current, setCurrent] = useState("");
   const navigate = useNavigate();
-
-  const [isLoggedIn, setIsLoggedIn] = useState(false);
-  console.log("isLoggedIn", isLoggedIn);
-
-  const confirmLogin = async () => {
-    try {
-      // Include credentials to send cookies with the request
-      const response = await fetch("/auth/confirm", {
-        credentials: "include",
-        headers: {
-          "Cache-Control": "no-cache",
-        },
-      });
-      if (response.status === 200) {
-        // If the backend returns a 200 status, the user is logged in
-        setIsLoggedIn(true);
-        console.log(response.status);
-      } else {
-        // If the response is not ok, set logged in to false
-        setIsLoggedIn(false);
-      }
-    } catch (error) {
-      console.error("Error confirming login", error);
-      setIsLoggedIn(false);
-    }
-  };
+  const { confirmLogin, isLoggedIn } = useAuth();
 
   useEffect(() => {
     confirmLogin();
-  }, [navigate, setIsLoggedIn, setCurrent]);
-
-  const onClick = (e) => {
-    setCurrent(e.key);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
This isn't really a comment, It's disabling an eslint warning. The warning wanted me to use confrimLogin in the dependency array of the useEffect hook. But I only wanted that hook to run once when the header first renders. Which is what the empty dependency array does.

A lot of this will be changed when i replace `confirmLogin` with `setIsLoggedIn` in the login and register components

Reply:
Ok ya gotcha I see